src/server/service/kv.rs

```diff
@@ -625,8 +627,27 @@ impl<T: RaftStoreRouter<RocksEngine> + 'static, E: Engine, L: LockManager> Tikv
                         my_store_id: store_id,
                     }))
                 } else {
-                    let ret = ch.send_raft_msg(msg).map_err(Error::from);
-                    future::ready(ret)
+                    let mut flag = true;
+                    if !disk::WRITE_PERMISSION.load(Ordering::Acquire) {
```

Comment:
It's better to move the check into `raftstore` before `step`.

Reply:
done